airflow/models/param.py

```diff
@@ -62,7 +62,9 @@ def _warn_if_not_json(value):
                 DeprecationWarning,
             )
 
-    def resolve(self, value: Any = NOTSET, suppress_exception: bool = False) -> Any:
+    def resolve(
+        self, value: Any = NOTSET, suppress_exception: bool = False, is_dag_parsing: bool = False
```

Comment:
Need to add this new argument to the docstring, and especially explain how it’s different from `suppress_exception`.

On a relative note, is there are more functional-descriptive name to this flag? Instead of `is_dag_parsing` (which means “this is set when DAG is being parsed”), something that describes how the flag changes behaviour, like how `suppress_exception` means “exceptions will not be raised”.